src/Backoffice/Airlines/Domain/Actions/ListAirlineAction.php

```diff
@@ -0,0 +1,24 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Lightit\Backoffice\Airlines\Domain\Actions;
+
+use Illuminate\Database\Eloquent\Model;
+use Illuminate\Pagination\LengthAwarePaginator;
+use Lightit\Backoffice\Airlines\Domain\Models\Airline;
+use Spatie\QueryBuilder\QueryBuilder;
+
+class ListAirlineAction
+{
+    /**
+     * @return LengthAwarePaginator<Model>
+     */
+    public function execute(): LengthAwarePaginator
+    {
+        return QueryBuilder::for(Airline::class)
+            ->allowedFilters(['name'])
+            ->allowedSorts('name')
+            ->paginate(10);
```

Comment:
If you add the flights, then you can use the with (read more about this here: https://laravel.com/docs/12.x/eloquent-relationships#automatically-hydrating-parent-models-on-children)
And also pay attention to the way we make de query, lets use directly the model instead of QueryBuilder. And by doing this, you can get rid of allowedFilters and allowedSorts because they are specific to laravel query builder
```suggestion
        return Airline::with(['flights'])
            ->paginate(10);
```

Reply:
I'm going to fix this in the other PR, where I modify the filtering.